src/Build/BackEnd/Components/SdkResolution/DefaultSdkResolver.cs

```diff
@@ -36,7 +37,7 @@ public override SdkResultBase Resolve(SdkReference sdk, SdkResolverContextBase c
             // Note: On failure MSBuild will log a generic message, no need to indicate a failure reason here.
             return FileUtilities.DirectoryExistsNoThrow(sdkPath)
                 ? factory.IndicateSuccess(sdkPath, string.Empty)
-                : factory.IndicateFailure(null);
+                : factory.IndicateFailure(null, new List<string>() { ResourceUtilities.FormatResourceStringStripCodeAndKeyword("DefaultSDKResolverError", sdk.Name, sdkPath) });
```

Comment:
Also nit: no need to pull in `System.Collections.Generic.List<T>`; you can pass a `string[]` as an `IEnumerable<string>`.